testenv/azure/asset.go

```diff
@@ -0,0 +1,201 @@
+		virtualNetworkName,
+		subnetName,
+		armnetwork.Subnet{
+			Properties: &armnetwork.SubnetPropertiesFormat{
+				AddressPrefix: to.Ptr("10.1.0.0/24"),
+			},
+		},
+		nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to begin creating subnet: %w", err)
+	}
+
+	resp, err := pollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create subnet: %w", err)
+	}
+
+	return &resp.Subnet, nil
+}
+
+func (e *AzureEnv) createNIC(ctx context.Context, location, resourceGroupName, subnetID string) (*armnetwork.Interface, error) {
+	pollerResp, err := e.interfacesClient.BeginCreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		networkInterfaceName,
+		armnetwork.Interface{
+			Location: to.Ptr(location),
+			Properties: &armnetwork.InterfacePropertiesFormat{
+				IPConfigurations: []*armnetwork.InterfaceIPConfiguration{
+					{
+						Name: to.Ptr("ipConfig"),
+						Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
+							PrivateIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodDynamic),
+							Subnet: &armnetwork.Subnet{
+								ID: to.Ptr(subnetID),
+							},
+						},
+					},
+				},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to begin creating network interface: %w", err)
+	}
+
+	resp, err := pollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create network interface: %w", err)
+	}
+	return &resp.Interface, nil
+}
+
+func (e *AzureEnv) createVirtualMachine(ctx context.Context, location, resourceGroupName, networkInterfaceID string) (*armcompute.VirtualMachine, error) {
+	parameters := armcompute.VirtualMachine{
+		Location: to.Ptr(location),
+		Tags:     map[string]*string{"scanconfig": to.Ptr("test")},
+		Identity: &armcompute.VirtualMachineIdentity{
+			Type: to.Ptr(armcompute.ResourceIdentityTypeNone),
+		},
+		Properties: &armcompute.VirtualMachineProperties{
+			StorageProfile: &armcompute.StorageProfile{
+				ImageReference: &armcompute.ImageReference{
+					// search image reference
+					// az vm image list --output table
+					Offer:     to.Ptr("0001-com-ubuntu-minimal-focal"),
+					Publisher: to.Ptr("Canonical"),
+					SKU:       to.Ptr("minimal-20_04-lts"),
+					Version:   to.Ptr("20.04.202004230"),
+				},
+				OSDisk: &armcompute.OSDisk{
+					Name:         to.Ptr(osDiskName),
+					CreateOption: to.Ptr(armcompute.DiskCreateOptionTypesFromImage),
+					ManagedDisk: &armcompute.ManagedDiskParameters{
+						StorageAccountType: to.Ptr(armcompute.StorageAccountTypesStandardSSDLRS), // OSDisk type Standard/Premium HDD/SSD
+					},
+				},
+			},
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes("Standard_B1ls")), // VM size include vCPUs,RAM,Data Disks,Temp storage.
+			},
+			OSProfile: &armcompute.OSProfile{ //
+				ComputerName:  to.Ptr("vmclarity-asset-compute"),
+				AdminUsername: to.Ptr("ubuntu"),
+				AdminPassword: to.Ptr("Password01!@#"),
```

Comment:
should this be specified?

Reply:
These are required fields I'm afraid...

Reply:
we should be able to do it without having to specify the password

Reply:
yes...

Reply:
This asset VM doesn't have a public IP though

Reply:
```go
			OSProfile: &armcompute.OSProfile{ // use username/password
				ComputerName:  to.Ptr("vmclarity-asset-compute"),
				AdminUsername: to.Ptr("vmclarity"),
				LinuxConfiguration: &armcompute.LinuxConfiguration{
					DisablePasswordAuthentication: to.Ptr(true),
				},
			},
```

Reply:
I added ssh auth instead of username/password!

Reply:
afaik we dont even need to be able to log into the VM since we are just going to create a snapshot and run the scan on it